lxd/endpoints/endpoints.go

```diff
@@ -210,6 +214,12 @@ func (e *Endpoints) up(config *Config) error {
 		return err
 	}
 
+	// Start the VM sock listener
+	e.listeners[vsock], err = createVsockListener(e.cert)
+	if err != nil {
+		logger.Warn("Failed creating VM socket listener", logger.Ctx{"err": err})
```

Comment:
This feels like it should return an error here, as that is a pretty bad scenario. If there's a specific reason why it shouldnt though please can you add a comment. Thanks

Reply:
I didn't make this an error, because LXD doesn't require VM/qemu/vsock to work.

Reply:
I see makes sense thanks.

Reply:
I wonder if we should tie this into the supported instance types logic so that we don't try and create the listener unless the host supports VMs? This would save the log spam when running on a host that only supports containers.